### DISCUSSION TITLE
Feature request [UI]: Remember column selection PER project in Traces view

### DISCUSSION THREAD

**Discussion by justingodden** [langfuse/langfuse#2720] [Ideas]
### Describe the feature or potential improvement

Hi, if I change the selected columns (below) when viewing traces, it will show the same view for every project.

![image](https://github.com/user-attachments/assets/9cbfd859-4fb3-4637-a717-e0d39408fd55)

But I have different things I want to look at for different projects (e.g. one project uses tags heavily, and another uses metadata).

I would like for LangFuse UI to remember the column configuration for each project individually.

Thanks!

**Comment by marcklingen** [comment]
thanks for sharing this, you are right, currently this configuration is saved globally (localstorage)
![CleanShot 2024-07-24 at 10 42 11](https://github.com/user-attachments/assets/895ddde0-b847-4f68-850a-2fa7c9e090a7)

**Comment by justingodden** [comment]
Could make it a nested object with project_id or project_name as key and the current object you are using as the value?

```js
tracesColumnVisibility = {
    project_1: {
        input: true,
        ...,
        version: false
    },
    project_2: {
        input: false,
        ...,
        version: true
    },
}
```

**Reply by marcklingen** [reply]
agree, this will be added in a work in progress change by @marliessophie
